packages/gatsby/cache-dir/production-app.js

```diff
@@ -85,13 +89,59 @@ apiRunnerAsync(`onClientEntry`).then(() => {
                   />
                 </Router>
               </ScrollContext>
+              <RouteAnnouncer />
             </RouteUpdates>
           )}
         </EnsureResources>
       )
     }
   }
 
+  const RouteAnnouncer = () => {
+    const [announcement, setAnnouncement] = useState(``)
+    useEffect(() => {
```

Comment:
should `useEffect` run on every update or it can be only for example when `location` is changed?

Reply:
I think ultimately we would need to use old lifecycle methods anyway (and not hooks) because we declare support for `react@^16.4.2` (which is few versions before hooks were introduced):
https://github.com/gatsbyjs/gatsby/blob/5d7825845296a33fa42bdd762fdc5e1b98f559bc/packages/gatsby/package.json#L183-L186

That's why we don't use any hooks in gatsby runtime right now (which is a bummer).

While this is in work in progress, I don't want to focus on implementation too much yet and first focus on high level ideas/questions:
 - use live-region to do announcements
 - use short content for page change announcements (title, h1, etc) instead of full new page content
 - how users will be able to customize the behaviour (and what should they be able to customize)

Reply:
what depends to push react dependency to current react version?